sync/src/main/kotlin/gropius/sync/AbstractSync.kt

```diff
@@ -323,6 +330,79 @@ abstract class AbstractSync(
         updater.save(collectedSyncInfo.nodeRepository)
     }
 
+    /**
+     * Lookup a state by its ID
+     * @param id ID of the state
+     * @return State with the given ID or null if given null
+     */
+    private suspend fun lookupState(id: String?): IssueState? {
+        if (id != null) {
+            return collectedSyncInfo.neoOperations.findById<IssueState>(id)!!
+        }
+        return null
+    }
+
+    private suspend fun integrateLabelToState(
+        timelineItems: MutableList<TimelineItem>,
+        rawTimelineItems: List<TimelineItem>,
+        issue: Issue,
+        imsProject: IMSProject
+    ) {
+        rawTimelineItems.toList().forEach { timelineItem ->
+            var lastState: IssueState = issue.state().value
```

Comment:
maybe it is possible to split this inner part into 2-3 subfunctions?
because this is really long (and it seems splitting could be reasonable, e.g. one function to first calculate the state things)

Reply:
a0cfce13d00e30d0323c4845dab70e6c8d39ea0c